crates/rpc/src/v02/method/add_deploy_account_transaction.rs

```diff
@@ -45,7 +66,15 @@ pub async fn add_deploy_account_transaction(
             tx.constructor_calldata,
         )
         .await
-        .context("Sending Deploy Account Transaction to the gateway")?;
+        .map_err(|e| match e {
+            SequencerError::StarknetError(e)
+                if e.code == KnownStarknetErrorCode::UndeclaredClass.into() =>
+            {
+                AddDeployAccountTransactionError::ClassHashNotFound
+            }
```

Comment:
This was actually not done previously, so this is a bugfix?

Reply:
Good question -- does the sequencer return UndeclaredClass for unknown class hashes?

Reply:
It does for `get_class` with an unknown class hash - unsure about an actual deploy..